include/aws/common/cbor.h

```diff
@@ -0,0 +1,450 @@
+#ifndef AWS_COMMON_CBOR_H
+#define AWS_COMMON_CBOR_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/common.h>
+
+AWS_PUSH_SANE_WARNING_LEVEL
+AWS_EXTERN_C_BEGIN
+
+/**
+ * The element types use by peek type.
+ * An extension for cbor major type in RFC8949 section 3.1
+ * Major type 0 - AWS_CBOR_TYPE_UINT
+ * Major type 1 - AWS_CBOR_TYPE_NEGINT
+ * Major type 2 - AWS_CBOR_TYPE_BYTESTRING/AWS_CBOR_TYPE_INF_BYTESTRING_START
+ * Major type 3 - AWS_CBOR_TYPE_STRING/AWS_CBOR_TYPE_INF_STRING_START
+ * Major type 4 - AWS_CBOR_TYPE_ARRAY_START/AWS_CBOR_TYPE_INF_ARRAY_START
+ * Major type 5 - AWS_CBOR_TYPE_MAP_START/AWS_CBOR_TYPE_INF_MAP_START
+ * Major type 6 - AWS_CBOR_TYPE_TAG
+ * Major type 7
+ *  - 20/21 - AWS_CBOR_TYPE_BOOL
+ *  - 22 - AWS_CBOR_TYPE_NULL
+ *  - 23 - AWS_CBOR_TYPE_UNDEFINE
+ *  - 25/26/27 - AWS_CBOR_TYPE_DOUBLE
+ *  - 31 - AWS_CBOR_TYPE_BREAK
+ *  - result of value not supported.
+ */
+enum aws_cbor_element_type {
+    AWS_CBOR_TYPE_UINT,
+    AWS_CBOR_TYPE_NEGINT,
+    AWS_CBOR_TYPE_DOUBLE,
+    AWS_CBOR_TYPE_BYTESTRING,
+    AWS_CBOR_TYPE_STRING,
```

Comment:
trivial: weird to do BYTESTRING but then just STRING

maybe: BYTESTRING / TEXTSTRING to stick with the official terminology

or: BYTES / TEXT to keep it short